command-line-interface.md

```diff
@@ -0,0 +1,189 @@
+# Operations
+
+A conformant [runtime][] MUST provide an executable (called `funC` in the following examples).
+That executable MUST support commands with the following template:
+
+```sh
+$ funC [global-options] <COMMAND> [command-specific-options] <command-specific-arguments>
+```
+
+## Global options
+
+None are required, but the runtime MAY support options that start with at least one hyphen.
+Global options MAY take positional arguments (e.g. `--log-level debug`).
+Command names MUST NOT start with hyphens.
+The option parsing MUST be such that `funC <COMMAND>` is unambiguously an invocation of `<COMMAND>` (even for commands not specified in this document).
+If the runtime is invoked with an unrecognized command, it MUST exit with a nonzero exit code and MAY log a warning to stderr.
+Beyond the above rules, the behavior of the runtime in the presence of commands and options not specified in this document is unspecified.
+
+## Character encodings
+
+This API specification does not cover character encodings, but runtimes SHOULD conform to their native operating system.
+For example, POSIX systems define [`LANG` and related environment variables][posix-lang] for [declaring][posix-locale-encoding] [locale-specific character encodings][posix-encoding], so a runtime in an `en_US.UTF-8` locale SHOULD write its [state](#state) to stdout in [UTF-8][].
+
+## Commands
+
+### create
+
+[Create][create] a container from a [bundle directory][bundle].
```

Comment:
On Thu, Jul 28, 2016 at 01:01:02PM -0700, W. Trevor King wrote:
> Anyhow, I'll push those per-command templates shortly.

Hmm, I was working this up, but it just ends up being:

  $ funC [global-options] create [command-specific-options] <ID>
  $ funC [global-options] start [command-specific-options] <ID>
  $ funC [global-options] state [command-specific-options] <ID>
  $ funC [global-options] kill [command-specific-options] <ID>
  $ funC [global-options] delete [command-specific-options] <ID>

Is that really all that useful for you [1]?  We need the
[global-options] and [command-specific-options] bits because we allow
runtimes to add additional command line options as they see fit (as
long as invocations that only use commands and options defined in this
spec work as expected).

[1]: https://github.com/opencontainers/runtime-spec/pull/513/files#r72680783